server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServer.java

```diff
@@ -75,15 +76,47 @@
     private boolean compress;
     private int maxLineLength;
     private int inMemorySizeLimit;
-    private boolean plainAuthDisallowed;
     private int timeout;
     private int literalSizeLimit;
+    private AuthenticationConfiguration authenticationConfiguration;
 
     public static final int DEFAULT_MAX_LINE_LENGTH = 65536; // Use a big default
     public static final Size DEFAULT_IN_MEMORY_SIZE_LIMIT = Size.of(10L, Size.Unit.M); // Use 10MB as default
     public static final int DEFAULT_TIMEOUT = 30 * 60; // default timeout is 30 minutes
     public static final int DEFAULT_LITERAL_SIZE_LIMIT = 0;
 
+    public static class AuthenticationConfiguration {
+        private static final boolean REQUIRE_SSL_DEFAULT = true;
+        private static final boolean PLAIN_AUTH_ENABLED_DEFAULT = true;
+
+        public static AuthenticationConfiguration parse(HierarchicalConfiguration<ImmutableNode> configuration) {
+            boolean requireSSLConfig = configuration.getBoolean("auth.requireSSL", REQUIRE_SSL_DEFAULT);
+            boolean plainAuthEnabled = configuration.getBoolean("auth.plainAuthEnabled", fallbackPlainAuthEnabled(configuration));
+
+            boolean requireSSL = false;
+            if (requireSSLConfig && !plainAuthEnabled) {
+                requireSSL = true;
```

Comment:
Also I don't think requireSSL value depends on plainAuthEnabled

Reply:
I think we should depend on plainAuthDisallowed here instead (allow fallback). Maybe:
 ```java
boolean requireSSL = true;
if (requireSSLConfig == false || plainAuthDisallowed = false) {
    requireSSL = false;
}
```
Where plainAuthDisallowed get from config file (as it stands now).

Reply:
Please avoid variable re-allocations you fools!